### ISSUE TITLE
bug: Fast changing of tabs makes inconsistent tab selection with Vue in Android + low performance with Vue

### ISSUE THREAD

**Issue by nshejteri** [ionic-team/ionic-framework#24707]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

Very fast changing between two tabs makes situation that one tab is selected in the footer but the content of another tab is displayed on the content. So if you try to tap on the selected tab in footer nothing will happen until you tap on non selected tab in the footer and then you will get correct content of new selected tab.

(Tab 1 is selected but the content of Tab 2 is displayed)
![tabs-issue-1](https://user-images.githubusercontent.com/5508186/152392757-b50fd6f7-73ce-4b4d-8eb8-b7520eea15f9.png)

Also there is situation when both of tabs you were switching between are selected in footer. 

![tabs-issue-2](https://user-images.githubusercontent.com/5508186/152392807-069999cf-b95d-44bf-a48f-20b291094706.png)

Also if you have some extra content on the tab pages, load will be slower and this issue will happed easier.

### Expected Behavior

Selected tabs in the footer should match displayed content no matter how fast you are switching between tabs.

### Steps to Reproduce

Just run Ionic starter tabs project with Vue and Android

ionic start myApp tabs --type vue
ionic integrations enable capacitor
ionic build
ionic cap add android
ionic cap open android
ionic cap sync

Run app on device (mine was Android 8) and start fast switching between two tabs. You will need few tries to reproduce issue.
Just taping for few seconds and stop to see situation and in one of tries it will happen. 

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.1
   Ionic Framework : @ionic/vue 6.0.5

Capacitor:

   Capacitor CLI      : 3.4.0
   @capacitor/android : 3.4.0
   @capacitor/core    : 3.4.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.3
   native-run  : 1.5.0

System:

   NodeJS : v16.13.1
   npm    : 8.4.0
   OS     : Windows 10

**Comment by nshejteri** [comment]
I want to add that tabs page rendering is very slow with Vue on Android (did not test on iOS). Also I did some tests with a starter tabs project with Vue and Angular where I added a simple list of 10 ion-cards on tab 1 and tab 3 with console logs on events onIonViewDidEnter and onIonViewWillEnter. It is clear that Angular is much faster. On Vue there is a big delay and lags when you are changing tabs very fast. For example when you stop tapping on tabs, tabs continue to switch like it continues to complete tasks from the queue.  If content on the tabs has more elements, performance goes down very fast with Vue.

Here are the repositories for easier reproduction. 
Just run these commands and try on device:

ionic integrations enable capacitor
ionic build
ionic cap add android
ionic cap sync

https://github.com/nshejteri/tabsTestAngular
https://github.com/nshejteri/tabsTestVue
